### PR TITLE
Added more test coverage for services.

### PR DESCRIPTION
**Pull request by runishkumar**

### CODE REVIEW

File: server/.env.SAMPLE [hunk 1]
```diff
@@ -22,4 +22,7 @@ COMPANY_NAME = ""
 
 # Key for Open Weather API
 WEATHER_API_KEY = ""
-WEATHER_API_CITY_ID = ""
\ No newline at end of file
+WEATHER_API_CITY_ID = ""
+
+#Trivia config
+TRIVIA_API_URL = "http://numbersapi.com"
```
Comment (rocketeric) [inline server/.env.SAMPLE]: I'd prefer to put this in constants or default config. The whole implementation or the trivia widget is coupled to this site so there is no need to make this URL configurable.

File: server/src/service/newRelicService.js [hunk 2]
```diff
@@ -35,6 +37,51 @@ class NewRelicService {
...
+
+  buildLoadTimeQuery() {
+    return `SELECT average(duration) from Transaction WHERE appName = '${this.newrelicSite}' since 10 minutes ago`;
+  }
+
+  buildUniqueSessionsQuery() {
+    return `SELECT count(session) FROM PageView WHERE appName = '${this.newrelicSite}' SINCE 20 minutes ago COMPARE WITH 20 minutes ago`;
+  }
+
+  buildSuccessfulBookingsUrl() {
```
Comment (rocketeric) [inline server/src/service/newRelicService.js]: Should be `buildSuccessfulBookingsQuery`?

File: server/src/service/newRelicService.js [hunk 3]
```diff
@@ -35,6 +37,51 @@ class NewRelicService {
...
+      SELECT count(session)
+      FROM PageView
+      WHERE pageUrl like 'https://${this.newrelicSite}/booking/%/success%'
+        and pageUrl not like '%pay-later%'
+      SINCE 1 day ago
+      COMPARE WITH 1 day ago
+    `;
+  }
+
+  buildCliErrorsUrl() {
```
Comment (rocketeric) [inline server/src/service/newRelicService.js]: Should be `buildCliErrorsQuery`?

File: server/src/service/newRelicService.js [hunk 4]
```diff
@@ -35,6 +37,51 @@ class NewRelicService {
...
+    return `SELECT count(*) from TransactionError WHERE appName = '${this.newrelicCliSite}' SINCE 30 minutes ago`;
+  }
+
+  buildErrorBreakdownUrl() {
```
Comment (rocketeric) [inline server/src/service/newRelicService.js]: Should be `buildErrorBreakdownQuery`?

File: server/src/service/newRelicService.js [hunk 5]
```diff
@@ -35,6 +37,51 @@ class NewRelicService {
...
+    return `SELECT count(*) from TransactionError WHERE appName = '${this.newrelicSite}' SINCE 1 day ago FACET \`error.class\``;
+  }
+
+  buildWebsiteFunnelUrl() { // eslint-disable-line
```
Comment (rocketeric) [inline server/src/service/newRelicService.js]: Should be `buildWebsiteFunnelQuery`?

File: server/src/service/newRelicService.js [hunk 6]
```diff
@@ -99,9 +138,9 @@ class NewRelicService {
   }
 
   getCLIErrors() {
-    const nrql = 'SELECT count(*) from TransactionError WHERE appName = \'cli.campsy.de\' SINCE 30 minutes ago';
+    // const nrql = 'SELECT count(*) from TransactionError WHERE appName = \'cli.campsy.de\' SINCE 30 minutes ago';
```
Comment (rocketeric) [inline server/src/service/newRelicService.js]: Could be cleaned up.

File: server/src/service/newRelicService.js [hunk 7]
```diff
@@ -148,7 +183,8 @@ class NewRelicService {
       return this.insights;
     }
 
-    if (!this.config.accountId) {
+    // Lodash/isEmpty function requires accountId to be string
```
Comment (rocketeric) [inline server/src/service/newRelicService.js]: It sure does :+1:

File: server/tests/service/triviaService.test.js [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+      const url = triviaService.buildTriviaUrl(today);
+      const expectedUrl = 'http://example.org/9/6/date';
+      expect(url).toEqual(expectedUrl);
+    });
+
+    it('builds correct url with invalid input', () => {
+      const today = new Date();
+      const url = triviaService.buildTriviaUrl('invalid');
+      const expectedUrl = triviaService.buildTriviaUrl(today);
+      expect(url).toEqual(expectedUrl);
```
Comment (rocketeric) [inline server/tests/service/triviaService.test.js]: What is this supposed to test? Passing a date into a buildUrl function?

File: server/tests/service/triviaService.test.js [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+    let axiosMock = null;
+    const today = new Date();
+
+    beforeEach(() => {
+      triviaService = new TriviaService(validConfig);
+      axiosMock = new MockAdapter(axios);
+    });
+
+    it('fetch trivia', () => {
+      const expectedTrivia = 'I don\'t win at trivia. Trivia loses to me. ~Chuck';
```
Comment (rocketeric) [inline server/tests/service/triviaService.test.js]: chuck

File: server/tests/service/triviaService.test.js [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+      axiosMock.onGet(triviaService.buildTriviaUrl(today)).reply(200, expectedTrivia);
+
+      triviaService.getTodaysTrivia().then((response) => {
+        expect(response).toEqual(expect.objectContaining({
+          body: expectedTrivia,
+        }));
+
+        expect(response).toHaveProperty('updated');
```
Comment (rocketeric) [inline server/tests/service/triviaService.test.js]: For kicks, you could check that the value is a valid date. I usually do it like this:
`expect(new Date(result.updated)).not.toBe(constants.testing.invalidDateString);`

File: server/tests/service/weatherService.test.js [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+
+      weatherService.fetchWeatherInfo().then((response) => {
+        expect(response).toEqual(expect.objectContaining({
+          city: 'Test',
+          temp: 14,
+          icon: '011',
+          description: 'heavy rain',
+        }));
+
+        expect(response).toHaveProperty('updated');
```
Comment (rocketeric) [inline server/tests/service/weatherService.test.js]: Can check for valid date here as well as any other places I might've missed.

File: server/tests/service/newRelicService.test.js [hunk 12]
```diff
@@ -0,0 +1,447 @@
...
+        expect(resp).toEqual({ status: 'error' });
+      }).catch((err) => {
+        expect(err).toEqual({ error: true });
+      });
+    });
+  });
+
+  describe('#getTransactionErrors', () => {
+    let newRelicService = null;
+    // let insightsInstance = null;
```
Comment (rocketeric) [inline server/tests/service/newRelicService.test.js]: Could clean this up.

File: server/tests/service/newRelicService.test.js [hunk 13]
```diff
@@ -0,0 +1,447 @@
...
+      return newRelicService.getTransactionErrors().then((response) => {
+        const errorResponse = NewRelicService.errorHandler();
+        expect(response).toEqual(errorResponse);
+      });
+    });
+  });
+
+  describe('#getLoadTime', () => {
+    let newRelicService = null;
+    // let insightsInstance = null;
```
Comment (rocketeric) [inline server/tests/service/newRelicService.test.js]: Could clean this up.

File: server/tests/service/newRelicService.test.js [hunk 14]
```diff
@@ -0,0 +1,447 @@
...
+      return newRelicService.getUniqueSessions().then((response) => {
+        const errorResponse = NewRelicService.errorHandler();
+        expect(response).toEqual(errorResponse);
+      });
+    });
+  });
+
+  describe('#getSuccessfulBookings', () => {
+    let newRelicService = null;
+    // let insightsInstance = null;
```
Comment (rocketeric) [inline server/tests/service/newRelicService.test.js]: Yep

File: server/tests/service/newRelicService.test.js [hunk 15]
```diff
@@ -0,0 +1,447 @@
...
+      return newRelicService.getSuccessfulBookings().then((response) => {
+        const errorResponse = NewRelicService.errorHandler();
+        expect(response).toEqual(errorResponse);
+      });
+    });
+  });
+
+  describe('#getCLIErrors', () => {
+    let newRelicService = null;
+    // let insightsInstance = null;
```
Comment (rocketeric) [inline server/tests/service/newRelicService.test.js]: :scissors:

File: server/tests/service/newRelicService.test.js [hunk 16]
```diff
@@ -0,0 +1,447 @@
...
+      return newRelicService.getCLIErrors().then((response) => {
+        const errorResponse = NewRelicService.errorHandler();
+        expect(response).toEqual(errorResponse);
+      });
+    });
+  });
+
+  describe('#getErrorBreakdown', () => {
+    let newRelicService = null;
+    // let insightsInstance = null;
```
Comment (rocketeric) [inline server/tests/service/newRelicService.test.js]: :scissors:

File: server/tests/service/newRelicService.test.js [hunk 17]
```diff
@@ -0,0 +1,447 @@
...
+      return newRelicService.getErrorBreakdown().then((response) => {
+        const errorResponse = NewRelicService.errorHandler();
+        expect(response).toEqual(errorResponse);
+      });
+    });
+  });
+
+  describe('#getWebsiteFunnel', () => {
+    let newRelicService = null;
+    // let insightsInstance = null;
```
Comment (rocketeric) [inline server/tests/service/newRelicService.test.js]: :scissors:

File: server/tests/service/triviaService.test.js (hunk 8 above)
Comment (runishkumar) [reply server/tests/service/triviaService.test.js]: It is to test if you pass invalid date to buildTriviaUrl, it will build the url for today's date.

**Comment by rocketeric** [comment]
Please see if you can add an additional case or two to get 100% branch coverage on the newRelicService. Looks doable without too much hassle.

**Pull request merged**